src/main.cpp

```diff
@@ -74,24 +71,29 @@ int encode(Image &image, const std::array<std::uint8_t, 32> &password, const std
     }
 
     // Read the data
-    auto padded_data = new std::uint8_t[padded_size];
+    auto padded_data = std::make_unique<uint8_t[]>(padded_size);
     file.seekg(0, std::ios::beg);
-    file.read(reinterpret_cast<char*>(padded_data), size);
+    file.read(reinterpret_cast<char*>(padded_data.get()), size);
     file.close();
 
     // Pad the data (#PKCS7)
     std::uint8_t left = padded_size - size;
-    std::fill(padded_data + size, padded_data + size + left, left);
+    std::fill_n(padded_data.get() + size, left, left);
 
     // Pick a random offset inside the image to store the data
     std::uint32_t offset;
     Random random;
-    random.get(&offset, sizeof(offset));
+	if (!random.get(&offset, sizeof(offset)))
+	{
+		std::cout << "Unable to generate random number\n";
```

Comment:
```suggestion
		std::cerr << "Unable to generate random number" << std::endl;
```
Please output errors to standard error, and use `std::endl` to make sure that the buffer is flushed.